modules/javafx.controls/src/main/java/javafx/scene/control/skin/TextInputControlSkin.java

```diff
@@ -324,83 +326,91 @@ protected boolean computeValue() {
                 }
             });
         }
+    }
+
+    @Override
+    public void install() {
+        super.install();
 
-        // FIXME: JDK-8268877 - incorrectly wired handler on replacing skin
+        TextInputControl control = getSkinnable();
         if (control.getOnInputMethodTextChanged() == null) {
             inputMethodTextChangedHandler = this::handleInputMethodEvent;
             control.setOnInputMethodTextChanged(inputMethodTextChangedHandler);
         }
 
```

Comment:
for the record: this pattern is incorrect (no matter _where_ the skin implements it ;) - a skin should not set a singleton event handler on its control (see [JDK-8093590](https://bugs.openjdk.org/browse/JDK-8093590)).

Not sure how we should procede in this case: moving to install fixes the issue on face value (not overwriting a user-installed handler) but not the incorrect handler registration (should be replaced by an added listener as done in the fix of JDK-8093590) - the latter could be done in a follow-up issue, adding a code comment here?

Reply:
Thank you.  You are right - we can use addEventHandler().